formats/json/commonMain/src/kotlinx/serialization/json/JsonElementBuilders.kt

```diff
@@ -186,6 +195,41 @@ public fun JsonArrayBuilder.addJsonObject(builderAction: JsonObjectBuilder.() ->
 public fun JsonArrayBuilder.addJsonArray(builderAction: JsonArrayBuilder.() -> Unit): Boolean =
     add(buildJsonArray(builderAction))
 
+/**
+ * Adds the given JSON [element] and [elements] to a resulting JSON array.
+ *
+ * @return `true` if the list was changed as the result of the operation.
+ */
+public fun JsonArrayBuilder.addAll(element: JsonElement, vararg elements: JsonElement): Boolean {
+    return addAll(listOf(element) + elements)
+}
+
+/** Adds the given string [values] to a resulting JSON array. */
+@JvmName("addAllStrings")
+public fun JsonArrayBuilder.addAll(values: Collection<String?>): Boolean =
+    addAll(values.map(::JsonPrimitive))
+
+/** Adds the given boolean [values] to a resulting JSON array. */
+@JvmName("addAllBooleans")
+public fun JsonArrayBuilder.addAll(values: Collection<Boolean?>): Boolean =
+    addAll(values.map(::JsonPrimitive))
+
+/** Adds the given numeric [values] to a resulting JSON array. */
+@JvmName("addAllNumbers")
+public fun JsonArrayBuilder.addAll(values: Collection<Number?>): Boolean =
+    addAll(values.map(::JsonPrimitive))
+
+/** Adds the given string [values] to a resulting JSON array. */
+public fun JsonArrayBuilder.addAll(vararg values: String?): Boolean =
```

Comment:
I'm not so sure about adding these functions, as stdlib's `buildList/MutableList` do not have them either. It's nice addition to DSL though, but they can be easily replaced with `addAll(listOf(...))` (and upcoming collections literals in future)

Reply:
I see what you mean, but I think they make using the JSON Builder DSL noticeably easier and less busy.

I suspect that these functions would be made redundant when (if?) Kotlin gets type-unions. Then only one function would be required...

```kotlin
public fun JsonArrayBuilder.addAll(vararg values: String? | Boolean? | Number?): Boolean 
```

And it would allow for adding a mix of values, e.g. `addAll(1, "a", false)`

But yes, collection literals would also help.

Reply:
I understand your reasoning, but we still would like to have them removed, as other places (like JsonArray constructor or stdlib) do not feature them at all, and it doesn't make really big difference.

Reply:
Agreed, it's not a big difference. I've removed the varargs util functions.